# swagger_client
API integration layer for linked biological objects.   __Source:__ https://github.com/monarch-initiative/biolink-api/

This Python package is automatically generated by the [Swagger Codegen](https://github.com/swagger-api/swagger-codegen) project:

- API version: 0.1.1
- Package version: 1.0.0
- Build package: io.swagger.codegen.languages.PythonClientCodegen

## Requirements.

Python 2.7 and 3.4+

## Installation & Usage
### pip install

If the python package is hosted on Github, you can install directly from Github

```sh
pip install git+https://github.com/GIT_USER_ID/GIT_REPO_ID.git
```
(you may need to run `pip` with root permission: `sudo pip install git+https://github.com/GIT_USER_ID/GIT_REPO_ID.git`)

Then import the package:
```python
import swagger_client 
```

### Setuptools

Install via [Setuptools](http://pypi.python.org/pypi/setuptools).

```sh
python setup.py install --user
```
(or `sudo python setup.py install` to install the package for all users)

Then import the package:
```python
import swagger_client
```

## Getting Started

Please follow the [installation procedure](#installation--usage) and then run the following:

```python
from __future__ import print_function
import time
import swagger_client
from swagger_client.rest import ApiException
from pprint import pprint
# create an instance of the API class
api_instance = swagger_client.AssociationApi()
id = 'id_example' # str | 
fl_excludes_evidence = true # bool | If set, excludes evidence objects in response (optional)
subject = 'subject_example' # str | SUBJECT id, e.g. NCBIGene:84570, ZFIN:ZDB-GENE-050417-357. Includes inferred by default (optional)
map_identifiers = 'map_identifiers_example' # str | Prefix to map all IDs to. E.g. NCBIGene (optional)
object = 'object_example' # str | OBJECT id, e.g. HP:0011927. Includes inferred by default (optional)
evidence = 'evidence_example' # str | Object id, e.g. ECO:0000501 (for IEA; Includes inferred by default)                     or a specific publication or other supporting ibject, e.g. ZFIN:ZDB-PUB-060503-2.                      (optional)
subject_taxon = 'subject_taxon_example' # str | SUBJECT TAXON id, e.g. NCBITaxon:9606. Includes inferred by default (optional)
rows = 10 # int | number of rows (optional) (default to 10)
page = 1 # int | Page number (optional) (default to 1)
graphize = true # bool | If set, includes graph object in response (optional)

try:
    # Returns the association with a given identifier
    api_response = api_instance.get_association_object(id, fl_excludes_evidence=fl_excludes_evidence, subject=subject, map_identifiers=map_identifiers, object=object, evidence=evidence, subject_taxon=subject_taxon, rows=rows, page=page, graphize=graphize)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling AssociationApi->get_association_object: %s\n" % e)

```

## Documentation for API Endpoints

All URIs are relative to *https://localhost/api*

Class | Method | HTTP request | Description
------------ | ------------- | ------------- | -------------
*AssociationApi* | [**get_association_object**](docs/AssociationApi.md#get_association_object) | **GET** /association/{id} | Returns the association with a given identifier
*AssociationApi* | [**get_association_search**](docs/AssociationApi.md#get_association_search) | **GET** /association/find/ | Generalized search over complete corpus of associations
*AssociationApi* | [**get_association_search_0**](docs/AssociationApi.md#get_association_search_0) | **GET** /association/find/{subject_category}/ | Returns list of matching associations
*AssociationApi* | [**get_association_search_1**](docs/AssociationApi.md#get_association_search_1) | **GET** /association/find/{subject_category}/{object_category}/ | Returns list of matching associations
*AssociationApi* | [**get_associations_from**](docs/AssociationApi.md#get_associations_from) | **GET** /association/between/{subject}/{object} | Returns associations connecting two entities
*AssociationApi* | [**get_associations_from_0**](docs/AssociationApi.md#get_associations_from_0) | **GET** /association/from/{subject} | Returns list of matching associations starting from a given subject (source)
*AssociationApi* | [**get_associations_to**](docs/AssociationApi.md#get_associations_to) | **GET** /association/to/{object} | Returns list of matching associations pointing to a given object (target)
*BioentityApi* | [**get_allele_object**](docs/BioentityApi.md#get_allele_object) | **GET** /bioentity/allele/{id} | TODO Returns allele object
*BioentityApi* | [**get_anatomy_gene_associations**](docs/BioentityApi.md#get_anatomy_gene_associations) | **GET** /bioentity/anatomy/{id}/genes/ | TODO Returns associations between anatomical entity and genes
*BioentityApi* | [**get_anatomy_object**](docs/BioentityApi.md#get_anatomy_object) | **GET** /bioentity/anatomy/{id} | TODO Returns anatomical entity
*BioentityApi* | [**get_anatomy_phenotype_associations**](docs/BioentityApi.md#get_anatomy_phenotype_associations) | **GET** /bioentity/anatomy/{id}/phenotypes/ | TODO Returns associations between anatomical entity and phenotypes
*BioentityApi* | [**get_disease_anatomy_associations**](docs/BioentityApi.md#get_disease_anatomy_associations) | **GET** /bioentity/disease/{id}/anatomy/ | TODO Returns anatomical locations associated with a disease
*BioentityApi* | [**get_disease_function_associations**](docs/BioentityApi.md#get_disease_function_associations) | **GET** /bioentity/disease/{id}/function/ | TODO Returns biological functions associated with a disease
*BioentityApi* | [**get_disease_gene_associations**](docs/BioentityApi.md#get_disease_gene_associations) | **GET** /bioentity/disease/{id}/genes/ | Returns genes associated with a disease
*BioentityApi* | [**get_disease_model_associations**](docs/BioentityApi.md#get_disease_model_associations) | **GET** /bioentity/disease/{id}/models/ | Returns associations to models of the disease
*BioentityApi* | [**get_disease_model_taxon_associations**](docs/BioentityApi.md#get_disease_model_taxon_associations) | **GET** /bioentity/disease/{id}/models/{taxon} | Same as &#x60;/disease/&lt;id&gt;/models&#x60; but constrain models by taxon
*BioentityApi* | [**get_disease_object**](docs/BioentityApi.md#get_disease_object) | **GET** /bioentity/disease/{id} | TODO Returns disease object
*BioentityApi* | [**get_disease_phenotype_associations**](docs/BioentityApi.md#get_disease_phenotype_associations) | **GET** /bioentity/disease/{id}/phenotypes/ | Returns phenotypes associated with disease
*BioentityApi* | [**get_disease_substance_associations**](docs/BioentityApi.md#get_disease_substance_associations) | **GET** /bioentity/disease/{id}/substance/ | Returns substances associated with a disease
*BioentityApi* | [**get_disease_substance_associations_0**](docs/BioentityApi.md#get_disease_substance_associations_0) | **GET** /bioentity/substance/{id}/treats/ | Returns substances associated with a disease
*BioentityApi* | [**get_environment_object**](docs/BioentityApi.md#get_environment_object) | **GET** /bioentity/environment/{id} | TODO Returns environment entity
*BioentityApi* | [**get_environment_phenotype_associations**](docs/BioentityApi.md#get_environment_phenotype_associations) | **GET** /bioentity/environment/{id}/phenotypes/ | TODO Returns list of associations
*BioentityApi* | [**get_gene_disease_associations**](docs/BioentityApi.md#get_gene_disease_associations) | **GET** /bioentity/gene/{id}/diseases/ | Returns diseases associated with gene
*BioentityApi* | [**get_gene_expression_associations**](docs/BioentityApi.md#get_gene_expression_associations) | **GET** /bioentity/gene/{id}/expressed/ | TODO Returns expression events for a gene
*BioentityApi* | [**get_gene_function_associations**](docs/BioentityApi.md#get_gene_function_associations) | **GET** /bioentity/gene/{id}/function/ | Returns function associations for a gene
*BioentityApi* | [**get_gene_homolog_associations**](docs/BioentityApi.md#get_gene_homolog_associations) | **GET** /bioentity/gene/{id}/homologs/ | Returns homologs for a gene
*BioentityApi* | [**get_gene_interactions**](docs/BioentityApi.md#get_gene_interactions) | **GET** /bioentity/gene/{id}/interactions/ | Returns interactions for a gene
*BioentityApi* | [**get_gene_object**](docs/BioentityApi.md#get_gene_object) | **GET** /bioentity/gene/{id} | Returns gene object
*BioentityApi* | [**get_gene_phenotype_associations**](docs/BioentityApi.md#get_gene_phenotype_associations) | **GET** /bioentity/gene/{id}/phenotypes/ | Returns phenotypes associated with gene
*BioentityApi* | [**get_gene_publication_list**](docs/BioentityApi.md#get_gene_publication_list) | **GET** /bioentity/gene/{id}/pubs/ | TODO Returns expression events for a gene
*BioentityApi* | [**get_geneproduct_object**](docs/BioentityApi.md#get_geneproduct_object) | **GET** /bioentity/geneproduct/{id} | TODO Returns gene product object
*BioentityApi* | [**get_generic_associations**](docs/BioentityApi.md#get_generic_associations) | **GET** /bioentity/{id}/associations/ | Returns associations for an entity regardless of the type
*BioentityApi* | [**get_generic_object**](docs/BioentityApi.md#get_generic_object) | **GET** /bioentity/{id} | TODO Returns object of any type
*BioentityApi* | [**get_genotype_disease_associations**](docs/BioentityApi.md#get_genotype_disease_associations) | **GET** /bioentity/genotype/{id}/diseases/ | Returns diseases associated with a genotype
*BioentityApi* | [**get_genotype_gene_associations**](docs/BioentityApi.md#get_genotype_gene_associations) | **GET** /bioentity/genotype/{id}/genes/ | Returns genes associated with a genotype
*BioentityApi* | [**get_genotype_genotype_associations**](docs/BioentityApi.md#get_genotype_genotype_associations) | **GET** /bioentity/genotype/{id}/genotypes/ | Returns genotypes-genotype associations
*BioentityApi* | [**get_genotype_object**](docs/BioentityApi.md#get_genotype_object) | **GET** /bioentity/genotype/{id} | Returns genotype object
*BioentityApi* | [**get_genotype_phenotype_associations**](docs/BioentityApi.md#get_genotype_phenotype_associations) | **GET** /bioentity/genotype/{id}/phenotypes/ | Returns phenotypes associated with a genotype
*BioentityApi* | [**get_goterm_gene_associations**](docs/BioentityApi.md#get_goterm_gene_associations) | **GET** /bioentity/goterm/{id}/genes/ | TODO Returns associated genes
*BioentityApi* | [**get_goterm_object**](docs/BioentityApi.md#get_goterm_object) | **GET** /bioentity/goterm/{id} | TODO Returns GO class object
*BioentityApi* | [**get_goterm_phenotype_associations**](docs/BioentityApi.md#get_goterm_phenotype_associations) | **GET** /bioentity/goterm/{id}/phenotype/ | TODO Returns associated phenotypes
*BioentityApi* | [**get_literature_disease_associations**](docs/BioentityApi.md#get_literature_disease_associations) | **GET** /bioentity/literature/{id}/diseases/ | Returns associations between a lit entity and a disease
*BioentityApi* | [**get_literature_gene_associations**](docs/BioentityApi.md#get_literature_gene_associations) | **GET** /bioentity/literature/{id}/genes/ | Returns associations between a lit entity and a gene
*BioentityApi* | [**get_literature_genotype_associations**](docs/BioentityApi.md#get_literature_genotype_associations) | **GET** /bioentity/literature/{id}/genotypes/ | Returns associations between a lit entity and a genotype
*BioentityApi* | [**get_parent_object**](docs/BioentityApi.md#get_parent_object) | **GET** /bioentity/individual/{id} | TODO Returns individual
*BioentityApi* | [**get_parent_object_0**](docs/BioentityApi.md#get_parent_object_0) | **GET** /bioentity/investigation/{id} | TODO Returns investigation object
*BioentityApi* | [**get_pathway_gene_associations**](docs/BioentityApi.md#get_pathway_gene_associations) | **GET** /bioentity/pathway/{id}/genes/ | TODO Returns list of genes associated with a pathway
*BioentityApi* | [**get_pathway_object**](docs/BioentityApi.md#get_pathway_object) | **GET** /bioentity/pathway/{id} | TODO Returns pathway object
*BioentityApi* | [**get_pathway_participant_associations**](docs/BioentityApi.md#get_pathway_participant_associations) | **GET** /bioentity/pathway/{id}/participants/ | TODO Returns associations to participants (molecules, etc) for a pathway
*BioentityApi* | [**get_phenotype_anatomy_associations**](docs/BioentityApi.md#get_phenotype_anatomy_associations) | **GET** /bioentity/phenotype/{id}/anatomy/ | Returns anatomical entities associated with a phenotype
*BioentityApi* | [**get_phenotype_function_associations**](docs/BioentityApi.md#get_phenotype_function_associations) | **GET** /bioentity/phenotype/{id}/function/ | TODO Returns biological functions associated with a Phenotype
*BioentityApi* | [**get_phenotype_gene_associations**](docs/BioentityApi.md#get_phenotype_gene_associations) | **GET** /bioentity/phenotype/{id}/gene/{taxid}/ids | Returns gene ids for all genes for a particular phenotype in a taxon
*BioentityApi* | [**get_phenotype_gene_associations_0**](docs/BioentityApi.md#get_phenotype_gene_associations_0) | **GET** /bioentity/phenotype/{id}/genes/ | TODO Returns associated phenotypes
*BioentityApi* | [**get_phenotype_object**](docs/BioentityApi.md#get_phenotype_object) | **GET** /bioentity/phenotype/{id} | TODO Returns phenotype class object
*BioentityApi* | [**get_phenotype_phenotype_associations**](docs/BioentityApi.md#get_phenotype_phenotype_associations) | **GET** /bioentity/phenotype/{id}/phenotype/ | TODO Returns associated phenotypes
*BioentityApi* | [**get_pub_object**](docs/BioentityApi.md#get_pub_object) | **GET** /bioentity/literature/{id} | TODO Returns publication object
*BioentityApi* | [**get_sequence_feature_object**](docs/BioentityApi.md#get_sequence_feature_object) | **GET** /bioentity/sequence_feature/{id} | TODO Returns seqfeature
*BioentityApi* | [**get_substance_exposures**](docs/BioentityApi.md#get_substance_exposures) | **GET** /bioentity/substance/{id}/exposures/ | TODO Returns associations between a substance and related exposures
*BioentityApi* | [**get_substance_interactions**](docs/BioentityApi.md#get_substance_interactions) | **GET** /bioentity/substance/{id}/interactions/ | TODO Returns associations between given drug and interactions
*BioentityApi* | [**get_substance_object**](docs/BioentityApi.md#get_substance_object) | **GET** /bioentity/substance/{id} | TODO Returns substance entity
*BioentityApi* | [**get_substance_participant_in_associations**](docs/BioentityApi.md#get_substance_participant_in_associations) | **GET** /bioentity/substance/{id}/participant_in/ | Returns associations between an activity and process and the specified substance
*BioentityApi* | [**get_substance_relationships**](docs/BioentityApi.md#get_substance_relationships) | **GET** /bioentity/substance/{id}/substances/ | TODO Returns associations between a substance and other substances
*BioentityApi* | [**get_substance_role_associations**](docs/BioentityApi.md#get_substance_role_associations) | **GET** /bioentity/substance/{id}/roles/ | Returns associations between given drug and roles
*BioentityApi* | [**get_substance_target_associations**](docs/BioentityApi.md#get_substance_target_associations) | **GET** /bioentity/substance/{id}/targets/ | TODO Returns associations between given drug and targets
*BioentityApi* | [**get_variant_gene_associations**](docs/BioentityApi.md#get_variant_gene_associations) | **GET** /bioentity/variant/{id}/genes/ | Returns genes associated with a variant
*BioentityApi* | [**get_variant_genotype_associations**](docs/BioentityApi.md#get_variant_genotype_associations) | **GET** /bioentity/variant/{id}/genotypes/ | Returns genotypes associated with a variant
*BioentityApi* | [**get_variant_object**](docs/BioentityApi.md#get_variant_object) | **GET** /bioentity/variant/{id} | TODO Returns sequence variant entity
*BioentityApi* | [**get_variant_phenotype_associations**](docs/BioentityApi.md#get_variant_phenotype_associations) | **GET** /bioentity/variant/{id}/phenotypes/ | Returns phenotypes associated with a variant
*BioentitysetApi* | [**get_entity_set_associations**](docs/BioentitysetApi.md#get_entity_set_associations) | **GET** /bioentityset/associations/ | Returns compact associations for a given input set
*BioentitysetApi* | [**get_entity_set_graph_resource**](docs/BioentitysetApi.md#get_entity_set_graph_resource) | **GET** /bioentityset/graph/ | TODO Graph object spanning all entities
*BioentitysetApi* | [**get_entity_set_over_representation_analysis**](docs/BioentitysetApi.md#get_entity_set_over_representation_analysis) | **GET** /bioentityset/ora/ | TODO Over-representation analysis
*BioentitysetApi* | [**get_entity_set_over_representation_analysis_0**](docs/BioentitysetApi.md#get_entity_set_over_representation_analysis_0) | **GET** /bioentityset/ora/{object_category}/ | TODO Over-representation analysis
*BioentitysetApi* | [**get_entity_set_summary**](docs/BioentitysetApi.md#get_entity_set_summary) | **GET** /bioentityset/descriptor/counts/ | Summary statistics for objects associated
*BioentitysethomologsApi* | [**get_entity_set_homologs**](docs/BioentitysethomologsApi.md#get_entity_set_homologs) | **GET** /bioentityset/homologs/ | Returns homology associations for a given input set of genes
*BioentitysetslimmerApi* | [**get_entity_set_slimmer**](docs/BioentitysetslimmerApi.md#get_entity_set_slimmer) | **GET** /bioentityset/slimmer/{category} | Summarize a set of objects
*CamApi* | [**get_activity_collection**](docs/CamApi.md#get_activity_collection) | **GET** /cam/activity/ | Returns list of models
*CamApi* | [**get_instance**](docs/CamApi.md#get_instance) | **GET** /cam/instance/{id} | Returns list of matches
*CamApi* | [**get_model**](docs/CamApi.md#get_model) | **GET** /cam/model/{id} | Returns a complete model
*CamApi* | [**get_model_collection**](docs/CamApi.md#get_model_collection) | **GET** /cam/model/ | Returns list of ALL models
*CamApi* | [**get_model_collection_0**](docs/CamApi.md#get_model_collection_0) | **GET** /cam/model/properties/ | Returns list of all properties used across all models
*CamApi* | [**get_model_collection_1**](docs/CamApi.md#get_model_collection_1) | **GET** /cam/model/property_values/ | Returns list property-values for all models
*CamApi* | [**get_model_collection_2**](docs/CamApi.md#get_model_collection_2) | **GET** /cam/model/query/ | Returns list of models matching query
*CamApi* | [**get_model_contibutors**](docs/CamApi.md#get_model_contibutors) | **GET** /cam/instances/ | Returns list of all instances
*CamApi* | [**get_model_contibutors_0**](docs/CamApi.md#get_model_contibutors_0) | **GET** /cam/model/contributors/ | Returns list of all contributors across all models
*CamApi* | [**get_physical_interaction**](docs/CamApi.md#get_physical_interaction) | **GET** /cam/physical_interaction/ | Returns list of models
*EvidencegraphApi* | [**get_association_object**](docs/EvidencegraphApi.md#get_association_object) | **GET** /evidence/graph/{id} | Returns evidence graph object for a given association
*EvidencegraphApi* | [**get_association_object_0**](docs/EvidencegraphApi.md#get_association_object_0) | **GET** /evidence/graph/{id}/image | Returns evidence graph as a png
*GenomefeaturesApi* | [**get_features_within_resource**](docs/GenomefeaturesApi.md#get_features_within_resource) | **GET** /genome/features/within/{build}/{reference}/{begin}/{end} | Returns list of matches
*GraphApi* | [**get_edge_resource**](docs/GraphApi.md#get_edge_resource) | **GET** /graph/edges/from/{id} | Returns edges emanating from a node
*GraphApi* | [**get_node_resource**](docs/GraphApi.md#get_node_resource) | **GET** /graph/node/{id} | Returns a graph node
*IdentifiermapperApi* | [**get_prefix_collection**](docs/IdentifiermapperApi.md#get_prefix_collection) | **GET** /identifier/mapper/{source}/{target}/ | TODO maps a list of identifiers from a source to a target
*IdentifierprefixesApi* | [**get_prefix_collection**](docs/IdentifierprefixesApi.md#get_prefix_collection) | **GET** /identifier/prefixes/ | Returns list of prefixes
*IdentifierprefixesApi* | [**get_prefix_collection_0**](docs/IdentifierprefixesApi.md#get_prefix_collection_0) | **GET** /identifier/prefixes/contract/{uri} | Returns contracted URI
*IdentifierprefixesApi* | [**get_prefix_collection_1**](docs/IdentifierprefixesApi.md#get_prefix_collection_1) | **GET** /identifier/prefixes/expand/{id} | Returns expanded URI
*IndividualApi* | [**get_individual**](docs/IndividualApi.md#get_individual) | **GET** /individual/{id} | Returns list of matches
*IndividualApi* | [**get_pedigree**](docs/IndividualApi.md#get_pedigree) | **GET** /individual/pedigree/{id} | Returns list of matches
*MartApi* | [**get_mart_case_associations_resource**](docs/MartApi.md#get_mart_case_associations_resource) | **GET** /mart/case/{object_category}/{taxon} | Bulk download of case associations
*MartApi* | [**get_mart_disease_associations_resource**](docs/MartApi.md#get_mart_disease_associations_resource) | **GET** /mart/disease/{object_category}/{taxon} | Bulk download of disease associations
*MartApi* | [**get_mart_gene_associations_resource**](docs/MartApi.md#get_mart_gene_associations_resource) | **GET** /mart/gene/{object_category}/{taxon} | Bulk download of gene associations
*NlpannotateApi* | [**get_annotate**](docs/NlpannotateApi.md#get_annotate) | **GET** /nlp/annotate/{text} | Not yet implemented
*OntolApi* | [**get_information_content_resource**](docs/OntolApi.md#get_information_content_resource) | **GET** /ontol/information_content/{subject_category}/{object_category}/{subject_taxon} | Returns information content (IC) for a set of relevant ontology classes
*OntollabelerApi* | [**get_ontol_labeler_resource**](docs/OntollabelerApi.md#get_ontol_labeler_resource) | **GET** /ontol/labeler/ | Fetches a map from CURIEs/IDs to labels
*OntolsubgraphApi* | [**get_extract_ontology_subgraph_resource**](docs/OntolsubgraphApi.md#get_extract_ontology_subgraph_resource) | **GET** /ontol/subgraph/{ontology}/{node} | Extract a subgraph from an ontology
*OntolsubgraphApi* | [**get_test_me**](docs/OntolsubgraphApi.md#get_test_me) | **GET** /ontol/subgraph/testme | Extract a subgraph from an ontology
*OwlontologyApi* | [**get_dl_query**](docs/OwlontologyApi.md#get_dl_query) | **GET** /owl/ontology/dlquery/{query} | Placeholder - use OWLery for now
*OwlontologyApi* | [**get_dl_query_0**](docs/OwlontologyApi.md#get_dl_query_0) | **GET** /owl/ontology/sparql/{query} | Placeholder - use direct SPARQL endpoint for now
*PairsimApi* | [**get_pair_sim_jaccard_resource**](docs/PairsimApi.md#get_pair_sim_jaccard_resource) | **GET** /pair/sim/jaccard/{id1}/{id2}/ | Get pairwise similarity
*PubpubsApi* | [**get_foo**](docs/PubpubsApi.md#get_foo) | **GET** /pub/pubs/{term} | TODO Returns list of matches
*RelationusageApi* | [**get_relation_usage_between_resource**](docs/RelationusageApi.md#get_relation_usage_between_resource) | **GET** /relation/usage/between/{subject_category}/{object_category} | All relations used plus count of associations
*RelationusageApi* | [**get_relation_usage_pivot_label_resource**](docs/RelationusageApi.md#get_relation_usage_pivot_label_resource) | **GET** /relation/usage/pivot/label | Relation usage count for all subj x obj category combinations, showing label
*RelationusageApi* | [**get_relation_usage_pivot_resource**](docs/RelationusageApi.md#get_relation_usage_pivot_resource) | **GET** /relation/usage/pivot/ | Relation usage count for all subj x obj category combinations
*RelationusageApi* | [**get_relation_usage_resource**](docs/RelationusageApi.md#get_relation_usage_resource) | **GET** /relation/usage/ | All relations used plus count of associations
*SearchApi* | [**get_autocomplete**](docs/SearchApi.md#get_autocomplete) | **GET** /search/entity/autocomplete/{term} | Returns list of matching concepts or entities using lexical search
*SearchApi* | [**get_search_entities**](docs/SearchApi.md#get_search_entities) | **GET** /search/entity/{term} | Returns list of matching concepts or entities using lexical search
*VariationsetApi* | [**delete_variant_set_item**](docs/VariationsetApi.md#delete_variant_set_item) | **DELETE** /variation/set/{id} | Deletes variant set
*VariationsetApi* | [**get_analyze**](docs/VariationsetApi.md#get_analyze) | **GET** /variation/set/analyze/{id} | Returns list of matches
*VariationsetApi* | [**get_variant_set_item**](docs/VariationsetApi.md#get_variant_set_item) | **GET** /variation/set/{id} | Returns a variant set
*VariationsetApi* | [**get_variant_sets_archive_collection**](docs/VariationsetApi.md#get_variant_sets_archive_collection) | **GET** /variation/set/archive/{year}/{month}/{day}/ | Returns list of variant sets from a specified time period
*VariationsetApi* | [**get_variant_sets_collection**](docs/VariationsetApi.md#get_variant_sets_collection) | **GET** /variation/set/ | Returns list of variant sets
*VariationsetApi* | [**post_variant_sets_collection**](docs/VariationsetApi.md#post_variant_sets_collection) | **POST** /variation/set/ | Creates a new variant set
*VariationsetApi* | [**put_variant_set_item**](docs/VariationsetApi.md#put_variant_set_item) | **PUT** /variation/set/{id} | Updates a variant set


## Documentation For Models

 - [APageOfResults](docs/APageOfResults.md)
 - [AbstractPropertyValue](docs/AbstractPropertyValue.md)
 - [Allele](docs/Allele.md)
 - [AnnotationExtension](docs/AnnotationExtension.md)
 - [Association](docs/Association.md)
 - [AssociationPropertyValue](docs/AssociationPropertyValue.md)
 - [AssociationResults](docs/AssociationResults.md)
 - [BioObject](docs/BioObject.md)
 - [CDS](docs/CDS.md)
 - [Chromosome](docs/Chromosome.md)
 - [CompactAssociationSet](docs/CompactAssociationSet.md)
 - [Edge](docs/Edge.md)
 - [EntityReference](docs/EntityReference.md)
 - [Exon](docs/Exon.md)
 - [Gene](docs/Gene.md)
 - [Genotype](docs/Genotype.md)
 - [Graph](docs/Graph.md)
 - [NamedObject](docs/NamedObject.md)
 - [Node](docs/Node.md)
 - [PageOfVariantSets](docs/PageOfVariantSets.md)
 - [Publication](docs/Publication.md)
 - [Relation](docs/Relation.md)
 - [SearchResult](docs/SearchResult.md)
 - [Seq](docs/Seq.md)
 - [SequenceFeature](docs/SequenceFeature.md)
 - [SequenceLocation](docs/SequenceLocation.md)
 - [SequencePosition](docs/SequencePosition.md)
 - [Substance](docs/Substance.md)
 - [SummaryPropertyValue](docs/SummaryPropertyValue.md)
 - [SynonymPropertyValue](docs/SynonymPropertyValue.md)
 - [Taxon](docs/Taxon.md)
 - [Transcript](docs/Transcript.md)
 - [VariantSet](docs/VariantSet.md)


## Documentation For Authorization

 All endpoints do not require authorization.


## Author



